docs/docs-beta/docs/guides/external-systems/non-python.md

```diff
@@ -1,4 +1,141 @@
 ---
 title: "Running non-Python languages"
 sidebar_position: 60
----
\ No newline at end of file
+---
+
+Dagster is written in Python, but that doesn't mean it's limited to running Python to materialize assets. This guide covers how to run some JavaScript with Dagster using Pipes.
+
+# Define your JavaScript function
+
+Here's a contrived example of some JavaScript code to run using Dagster Pipes. `train_model` load some dataset and train a sequential model using tensorflow.
+
+```javascript
+import * as tf from '@tensorflow/tfjs';
+
+async function train_model(config) {
+  const { path_to_data, data_config, path_to_model } = config;
+  const dataset = await tf.data.csv(path_to_data, data_config).map(({ xs, ys }) => {
+    return {
+      xs: tf.tensor2d(Object.values(xs), [Object.values(xs).length, 1]),
+      ys: tf.tensor2d(Object.values(ys), [Object.values(ys).length, 1])
+    };
+  })
+  .batch(100);
+
+  const model = tf.sequential()
+  model.add(tf.layers.dense({units: 1, inputShape: [1]}));
+  model.compile({loss: 'meanSquaredError', optimizer: 'sgd'});
+
+  await model.fitDataset(dataset, {epochs: 250})
+  await model.save(path_to_model);
+  model.summary();
+  return model;
+}
+```
+
+# Define your asset in Dagster
+
+<CodeExample filePath="guides/automation/pipes-asset.py" language="python" title="Asset using Dagster Pipes." />
+
+Dagster Pipes follows a similar design to Unix pipes, hence the name. The `PipesSubprocessClient` is responsible for running external processes and setting up input/output files. The asset defined here is materialized using the `PipesSubprocessClient` running a Node.js file containing the `train_model` function.
+
```

Comment:
i think here at this step, we can invoke the js script already.

one confusion around pipes is that one has to make changes to their external script in order to make pipes work. but in fact, you don't have to.

so i'd suggest adding a section here - something like "Run Typescript from Dagster without code changes"

without external code changes (any js util functions to access pipes), you still get:
- know when the job completed or failed (subprocess will exit 1)
- get stdout/err from node/subprocess to dagster

but you won't get:
- pass vars from dagster to typescript
- stream logs from databricks to dagster

then, the following sections can be wrapped up as an "advanced: pass data between dagster and typescript"